### PR TITLE
Fix for the lacking generic lambda support in visual studio

### PR DESCRIPTION
**Pull request by WolfgangVogl**
This this fix it compiles in visual studio 15.9.x.
Minded the discussion in  the other pull request.

### CODE REVIEW

File: include/belle.hh [hunk 1]
```diff
@@ -1196,8 +1196,13 @@ private:
 
   private:
 
+#ifdef _MSC_VER
+    template <typename SELF_T, typename RES_T>
+    static constexpr void send(SELF_T self, RES_T&& res)
```
Comment (octobanana) [inline include/belle.hh]: If you don't mind, I'd prefer to use the template names `Self` and `Res`. I can make the change when merging if that is alright with you.

**Comment by octobanana** [comment]
Apologies for the delay in getting back to you, If you're alright with the naming changes I can merge this in.

File: include/belle.hh (hunk 1 above)
Comment (WolfgangVogl) [reply include/belle.hh]: I commited the changes!

**Comment by octobanana** [comment]
merged in bc7b6450f15181ec848ed35ba11ac5b2cf2f1ae5

**Pull request closed without merging**